scripts/quests/otherAreas/Knocking_on_Forbidden_Doors.lua

```diff
@@ -0,0 +1,277 @@
+                end,
+            },
+        },
+    },
+
+    -- QUEST ACCEPTED
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.TAVNAZIAN_SAFEHOLD] =
+        {
+            ['Chemioue'] =
+            {
+                onTrigger = function(player, csid, option, npc)
+                    if quest:getVar(player, 'prog') == 0 then
+                        return quest:event(536)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [536] = function(player, csid, option, npc)
+                    quest:setVar(player, 'prog', 1)
+                end,
+            },
+        },
+
+        [xi.zone.PHOMIUNA_AQUEDUCTS] =
+        {
+            ['Wooden_Ladder'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'prog') == 1 and
+                    npc:getID() == ID.npc.LADDER_KNOCKING then -- Find a better way to identify exactly which ladder it is in case IDs shift
+                        return quest:progressEvent(38, 27) -- Essentially chains the ladder cs with the actual quest cs
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [38] = function(player, csid, option, npc)
+                    quest:setVar(player, 'prog', 2)
+                end,
+            },
+        },
+
+        [xi.zone.MISAREAUX_COAST] =
+        {
+            ['qm_mire_incense'] =
+            {
+                onTrigger = function(player, npc)
+                    --- ??? by end of river
+                    if quest:getVar(player, 'prog') == 2 and
+                    not player:hasKeyItem(xi.ki.MIRE_INCENSE) then
+                        player:addKeyItem(xi.ki.MIRE_INCENSE)
+                        return quest:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.MIRE_INCENSE)
+                    end
+                end,
+            },
+
+            ['qm_alsha'] =
+            {
+                onTrigger = function(player, npc)
+                    -- First time clicking on ???
+                    if player:hasKeyItem(xi.ki.MIRE_INCENSE) and
+                    quest:getVar(player, 'prog') == 2 then
+                        return quest:progressEvent(556, { [1] = xi.ki.MIRE_INCENSE })
+
+                    -- Second time clicking on ???
+                    elseif quest:getVar(player, 'prog') == 3 then
+                        return quest:progressEvent(557, { [1] = xi.ki.MIRE_INCENSE })
+
+                    -- Clicking on the ??? after killing NM
+                    elseif quest:getVar(player, 'prog') == 4 then
+                        return quest:event(558)
+                    end
+                end,
+            },
+
+            ['Alsha'] =
+            {
+                onMobDeath = function(mob, player, isKiller, firstCall)
```

Comment:
`onMobDeath = function(mob, player, optParams)`  Function and logic needs to be updated